app-example/src/main/java/com/badoo/ribs/example/root/RootBuilder.kt

```diff
@@ -0,0 +1,71 @@
+package com.badoo.ribs.example.root
+
+import com.badoo.ribs.builder.SimpleBuilder
+import com.badoo.ribs.core.modality.BuildParams
+import com.badoo.ribs.example.auth.AuthDataSource
+import com.badoo.ribs.example.auth.AuthDataSourceImpl
+import com.badoo.ribs.example.root.routing.RootChildBuilders
+import com.badoo.ribs.example.root.routing.RootRouter
+import com.badoo.ribs.example.root.routing.RootRouter.Configuration
+import com.badoo.ribs.routing.source.RoutingSource
+import com.badoo.ribs.routing.source.backstack.BackStackFeature
+
+class RootBuilder(
+    private val dependency: Root.Dependency
+) : SimpleBuilder<Root>() {
+
+    override fun build(buildParams: BuildParams<Nothing?>): Root {
+        val authDataSource = authDataSource(dependency)
+        val connections = RootChildBuilders(dependency, authDataSource)
+        val customisation = buildParams.getOrDefault(Root.Customisation())
+        val backStack = backStack(buildParams)
+        val interactor = interactor(buildParams, backStack, authDataSource)
+        val router = router(buildParams, backStack, connections, customisation)
+
+        return node(buildParams, interactor, router)
```

Comment:
I'd inline this, and the other methods too. They were methods only for the sake of Dagger Provides blocks.